src/lib.rs

```diff
@@ -0,0 +1,55 @@
+#[macro_use]
```

Comment:
depending on your pedandticness, I usually put:

```
#![deny(warnings, missing_debug_implementations, missing_copy_implementations)]
```

at the top of crates (lib.rs or main,rs) for linting purposes.

Reply:
My pedanticness is quite high, but I think I'd rather just derive implementations for those. Is there a reason not to?

Reply:
this doesn't force you to derive, it causes the compiler to error if you don't do a derive (or your own implementation)

Reply:
Oh, I see, that was the opposite of what I thought it was doing! In that case yes, sure, I'll add that. Thanks.